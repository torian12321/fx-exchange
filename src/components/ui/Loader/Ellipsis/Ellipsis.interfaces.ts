export interface IEllipsis {
  children?: never;
  className?: string;
}
