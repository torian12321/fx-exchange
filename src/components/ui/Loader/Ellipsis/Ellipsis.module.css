.wrapp {
  font-size: 12px;
  color: #b159a4;
}

.dot {
  display: inline-block;
  width: .8em;
  height: .8em;
  border-radius: 100%;
  background: currentColor;
  margin: 0 .2em;
  opacity: 0;
}

.dot:nth-child(1) {
  animation: opacitychange 1s ease-in-out infinite;
}

.dot:nth-child(2) {
  animation: opacitychange 1s ease-in-out 0.33s infinite;
}

.dot:nth-child(3) {
  animation: opacitychange 1s ease-in-out 0.66s infinite;
}

@keyframes opacitychange {
  0%, 100%{
    opacity: .2;
  }

  60%{
    opacity: 1;
  }
}