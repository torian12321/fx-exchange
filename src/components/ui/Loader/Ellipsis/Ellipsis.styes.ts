import { css, keyframes } from '@emotion/core'
import { theme } from 'assets/styles';

const opacitychange = keyframes`
  0%, 100% { opacity: .2; }
  60%      { opacity: 1; }
`

export const wrapp = css`
  font-size: 12px;
  color: ${theme.colors.secondary};
`

export const dot = css`
  display: inline-block;
  width: .8em;
  height: .8em;
  border-radius: 100%;
  background: currentColor;
  margin: 0 .2em;
  opacity: 0;

  animation: ${opacitychange} 1s ease infinite;

  &:nth-child(1) { animation-delay: 0; }
  &:nth-child(2) { animation-delay: 0.33s; }
  &:nth-child(3) { animation-delay: 0.66s; }
`