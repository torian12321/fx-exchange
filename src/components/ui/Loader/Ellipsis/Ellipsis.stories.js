import React from 'react';
import { Ellipsis } from './Ellipsis';

export default {
  title: 'UI Components|Loaders/Ellipsis',
};

export const Basic = () => <Ellipsis />;
