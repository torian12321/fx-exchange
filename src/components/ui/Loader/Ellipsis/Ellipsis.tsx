import React from "react";
/** @jsx jsx */
import { jsx } from '@emotion/core'
import { IEllipsis } from './Ellipsis.interfaces';
import * as styles from './Ellipsis.styes'

const Ellipsis = ({ className }: IEllipsis) => (
  <div css={styles.wrapp} className={className}>
    <span css={styles.dot} />
    <span css={styles.dot} />
    <span css={styles.dot} />
  </div>
);

export { Ellipsis };
