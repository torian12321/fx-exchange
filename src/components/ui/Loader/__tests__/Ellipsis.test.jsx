import React from "react";
import { create } from "react-test-renderer";
import { Ellipsis } from "../Ellipsis";

describe("UI Component: Loader -> Ellipsis", () => {
  it(`should render with 3 spans`, () => {
    const component = create(
      <Ellipsis />
    );

    const ellipsis = component.root;

    expect(ellipsis.findAllByType("div").length).toEqual(1);
    expect(ellipsis.findAllByType("span").length).toEqual(3);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
