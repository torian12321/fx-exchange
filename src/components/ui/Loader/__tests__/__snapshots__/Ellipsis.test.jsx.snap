// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Component: Loader -> Ellipsis should render with 3 spans 1`] = `
<div
  className=" css-1pmieoy-Ellipsis"
>
  <span
    className="css-wraxhb-Ellipsis"
  />
  <span
    className="css-wraxhb-Ellipsis"
  />
  <span
    className="css-wraxhb-Ellipsis"
  />
</div>
`;
